File: class2relational/src/main/java/org/atlanmod/model/dynamic/classModel/AttributeToClass.java
package org.atlanmod.model.dynamic.classModel;

import org.atlanmod.model.DynamicElement;

import java.util.ArrayList;
import java.util.List;

public class AttributeToClass extends ClassLink {

    public AttributeToClass(ClassAttribute source, ClassClass target) {
        super(ClassMetamodel.ATTRIBUTE_CLASS, source, null);
        List<DynamicElement> new_target = new ArrayList<>();
        new_target.add(target);
        this.setTarget(new_target);
    }

    public String toString() {
        return "(" +
                ((ClassElement) this.getSource()).getId() +
                ", " + super.getType() +
                ", " + ((ClassElement) this.getTarget().get(0)).getId() +
                ")";
    }
}

File: relational2class/src/main/scala/org/atlanmod/model/classmodel/AttributeToClass.scala
package org.atlanmod.model.classmodel

class AttributeToClass(source: ClassAttribute, target: ClassClass)
  extends ClassLink(ClassMetamodel.ATTRIBUTE_CLASS, source, List(target)) {

    override def toString: String =
        "(" + source.getId() + ", " + super.getType + ", " + getTargetClass.getId() + ")"

    override def getSource: ClassAttribute = source

    override def getTarget: List[ClassClass] = List(target)

    def getTargetClass: ClassClass = target

}
